airbyte-integrations/connectors/destination-mssql/src/test-integration/kotlin/io/airbyte/integrations/destination/mssql/v2/MSSQLWriterTest.kt

```diff
@@ -66,7 +66,8 @@ abstract class MSSQLWriterTest(
         preserveUndeclaredFields = false,
         supportFileTransfer = false,
         commitDataIncrementally = true,
-        allTypesBehavior = StronglyTyped(integerCanBeLarge = false),
+        allTypesBehavior =
+            StronglyTyped(integerCanBeLarge = false, nestedFloatLosesPrecision = false),
```

Comment:
this breaks BulkInsert.testBasicTypes, because bulk insert actually loses precision on nested floats. That's fixed by [this](https://github.com/airbytehq/airbyte/pull/55884/files#diff-ebab2eed4db9b508776175e26c3c9cca724b193ac1fe701e0ab46714a7a8e218R106), i.e. instead of recursing through the object, we just serialize it wholesale.